### PR TITLE
YLP-1428 invite monitoring patient

### PR DESCRIPTION
**Pull request by franck-fourel**
- add routes to manage invite/accept/dismiss monitoring patient
 - dependency on https://github.com/mdblp/crew/pull/36 so that crew client is updated with new minitoring features
 - little go dependencies update (there is one one to please testify that Im'n not sure to understand)

Still open questions on what we should do when a monitoring request is cancelled by a HCP: should we inform the patient. The current solution does not make any emails to the patient.

### CODE REVIEW

File: api/invite.go [hunk 1]
```diff
@@ -1200,6 +1481,91 @@ func (a *Api) SendTeamInvite(res http.ResponseWriter, req *http.Request, vars ma
...
+		res.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	patientid := vars["userid"]
+	teamid := vars["teamid"]
+
+	_, team, _ := a.getTeamForUser(tokenValue, teamid, token.UserId, res)
+
+	if team.ID == "" {
```
Comment (CedricPerotto) [inline api/invite.go:1530]: I think we should also check that
1. The current user is an admin of the team (As only admins can dismiss invites...)
2. The current team is a monitored team

File: api/invite_test.go [hunk 2]
```diff
@@ -218,10 +277,10 @@ func initTestingTeamRouter(returnNone bool) *mux.Router {
...
 		InvitationStatus: "pending",
 	}
-	patient_dismissed := store.Member{
-		UserID:           testing_uid4,
-		TeamID:           "123456",
-		Role:             "patient",
+	patient_dismissed := store.Patient{
+		UserID: testing_uid4,
+		TeamID: "123456",
+		// Role:             "patient",
```
Comment (CedricPerotto) [inline api/invite_test.go]:
```suggestion
```

File: templates/html/medicalteam_monitoring_invitation.html [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                    </td>
+                  </tr>
+                  <tr>
+                    <td class="inner centered" style="padding:0;padding:10px;text-align:center;">
+                      <!--[if (gte mso 9)|(IE)]>
+                        <table bgcolor="#627CFF">
+                          <tr>
+                            <td>
+                      <![endif]-->
+                      <a class="btn primary" href="{{.WebURL}}/{{ .WebPath }}?inviteEmail={{ .Email }}" style="text-decoration:none;display:inline-block;font-family:'Ubuntu', sans-serif;border-radius:4px;padding:10px 20px;background-color:#00a3e2;font-size:16px;font-weight:bold;color:#ffffff;margin-left:5px;margin-right:5px;margin-bottom:10px;">
```
Comment (CedricPerotto) [inline templates/html/medicalteam_monitoring_invitation.html:87]: I can't see this inviteEmail={{ .Email }} query parameter in blip code source

**Review by CedricPerotto — COMMENTED** [review]
See my comments as already discussed build.sh + **templates/preview/build.sh (for hydromail)** need to be updated with 
`go mod tidy -compat=1.17`

**Review by CedricPerotto — CHANGES_REQUESTED** [review]
Another global comment the `GetReceivedInvitations` API method (and maybe `GetSentInvitations`) should be updated to include TypeMedicalTeamMonitoringInvite types when role is patient (role is not used in `GetSentInvitations`).

**Review by ydrisR — APPROVED** [review]
LGTM

**Pull request merged**